mediator/med_map_mod.F90

```diff
@@ -594,529 +587,629 @@ logical function med_map_RH_is_created_RH1d(RHs,mapindex,rc)
+                      end if
+                   end do
+                else
+                   npacked(mapindex) = npacked(mapindex) + 1
+                   packed_data(mapindex)%fldindex(nf) = npacked(mapindex)
+                end if
 
-                used_cart3d_for_uvmapping = .true.
-             end if
-             CYCLE
-          end if
-       end if
+                if (mastertask) then
+                   write(logunit,'(5(a,2x),2x,i4)') trim(subname)//&
+                        'Packed field: destcomp,mapping,mapnorm,fldname,index: ', &
+                        trim(compname(destcomp)), &
+                        trim(mapnames(mapindex)), &
+                        trim(packed_data(mapindex)%mapnorm), &
+                        trim(fieldnamelist(nf)), &
+                        packed_data(mapindex)%fldindex(nf)
+                end if
 
-       ! -------------------
-       ! Get the source and destination fields
-       ! -------------------
+             end if! end if source field is mapped to destination field with mapindex
+          end do ! end loop over FBSrc fields
+       end do ! end loop over fldsSrc elements
 
-       call ESMF_LogWrite(trim(subname)//" --> remapping "//trim(fldname)//" with "//trim(mapnames(mapindex)), &
-            ESMF_LOGMSG_INFO)
+       if (npacked(mapindex) > 0) then
+          ! Create the packed source field bundle for mapindex
+          allocate(ptrsrc_packed(npacked(mapindex), lsize_src))
+          packed_data(mapindex)%field_src = ESMF_FieldCreate(lmesh_src, &
+               ptrsrc_packed, gridToFieldMap=(/2/),  meshloc=ESMF_MESHLOC_ELEMENT, rc=rc)
+          if (chkerr(rc,__LINE__,u_FILE_u)) return
 
-       call ESMF_FieldBundleGet(FBSrc, fieldName=trim(fldname), field=srcfield, rc=rc)
-       if (chkerr(rc,__LINE__,u_FILE_u)) return
-       call ESMF_FieldBundleGet(FBDst, fieldName=trim(fldname), field=dstfield, rc=rc)
-       if (chkerr(rc,__LINE__,u_FILE_u)) return
+          ! Create the packed destination field bundle for mapindex
+          allocate(ptrdst_packed(npacked(mapindex), lsize_dst))
+          packed_data(mapindex)%field_dst = ESMF_FieldCreate(lmesh_dst, &
+               ptrdst_packed, gridToFieldMap=(/2/),  meshloc=ESMF_MESHLOC_ELEMENT, rc=rc)
+          if (chkerr(rc,__LINE__,u_FILE_u)) return
 
-       ! -------------------
-       ! Do the mapping
-       ! -------------------
+          packed_data(mapindex)%field_fracsrc = ESMF_FieldCreate(lmesh_src, ESMF_TYPEKIND_R8, &
+               meshloc=ESMF_MESHLOC_ELEMENT, rc=rc)
+          packed_data(mapindex)%field_fracdst = ESMF_FieldCreate(lmesh_dst, ESMF_TYPEKIND_R8, &
+               meshloc=ESMF_MESHLOC_ELEMENT, rc=rc)
+       end if
+    end do ! end loop over mapindex
 
-       if (mapindex == mapfcopy) then
-          call med_map_FB_Field_Regrid(FBSrc, fldname, FBDst, fldname, RouteHandles, mapindex, rc=rc)
-          if (chkerr(rc,__LINE__,u_FILE_u)) return
+    deallocate(npacked)
```

Comment:
same for the deallocate statements. if those are already allocated then the deallocate can be called.

Reply:
You need this deallocate for npacked since you are allocating and deallocating it as you go through the mapindices. I don't ever see a case where you would trigger a deallocate for an unallocated array. I'll double check.